src/main/java/ch/njol/skript/classes/data/BukkitClasses.java

```diff
@@ -455,61 +462,60 @@ public OfflinePlayer parse(final String s, final ParseContext context) {
             }
 
             @Override
-            public boolean canParse(final ParseContext context) {
+            public final boolean canParse(final ParseContext context) {
                 return context == ParseContext.COMMAND;
             }
 
             @Override
-            public String toString(final OfflinePlayer p, final int flags) {
+            public final String toString(final OfflinePlayer p, final int flags) {
                 return p.getName();
             }
 
             @Override
-            public String toVariableNameString(final OfflinePlayer p) {
+            public final String toVariableNameString(final OfflinePlayer p) {
                 if (SkriptConfig.usePlayerUUIDsInVariableNames.value())
                     return p.getUniqueId().toString();
                 return p.getName();
             }
 
             @Override
-            public String getVariableNamePattern() {
+            public final String getVariableNamePattern() {
                 if (SkriptConfig.usePlayerUUIDsInVariableNames.value())
                     return "[0-9a-f]{8}(-[0-9a-f]{4}){3}-[0-9a-f]{12}";
                 return "\\S+";
             }
 
             @Override
-            public String getDebugMessage(final OfflinePlayer p) {
+            public final String getDebugMessage(final OfflinePlayer p) {
                 if (p.isOnline())
                     return Classes.getDebugMessage(p.getPlayer());
                 return p.getName();
             }
         }).serializer(new Serializer<OfflinePlayer>() {
-            private final boolean uuidSupported = Skript.methodExists(OfflinePlayer.class, "getUniqueId");
-
             @Override
-            public Fields serialize(final OfflinePlayer p) {
+            public final Fields serialize(final OfflinePlayer p) {
                 final Fields f = new Fields();
-                if (uuidSupported)
+                if (Skript.offlineUUIDSupported || dontUseNames)
                     f.putObject("uuid", p.getUniqueId());
-                f.putObject("name", p.getName());
```

Comment:
The actual fix is here that removes the extra un-needed OfflinePlayer#getName call which reads player data from the disk.